pipelines/common/src/main/java/org/openmrs/analytics/DwhFiles.java

```diff
@@ -115,32 +118,58 @@ public Path getIncrementalRunPathWithTimestamp() {
    * @return same as {@link #getIncrementalRunPath()}
    * @throws IOException if the directory move fails
    */
-  public Path newIncrementalRunPath() throws IOException {
-    Path incPath = getIncrementalRunPath();
+  public ResourceId newIncrementalRunPath() throws IOException {
```

Comment:
If it is not too much work, consider improving the unit-test coverage in this class. For example, I think this method  and `read/writeTimestampFile` (for local mode) do not have unit-tests.

Reply:
Test cases added